src/services/action/repositories/action.ts

```diff
@@ -167,12 +191,25 @@ export class ActionRepository {
       subquery.andWhere('action.type IN (:...types)');
     }
 
+    if (filters?.startDate) {
+      subquery.andWhere('action.created_at >= :startDate', {
+        startDate: filters.startDate,
+      });
+    }
+
+    if (filters?.endDate) {
+      subquery.andWhere('action.created_at <= :endDate', {
+        endDate: filters.endDate,
+      });
+    }
     // Second stage aggregation.
     const query = AppDataSource.createQueryBuilder()
       .from(`(${subquery.getQuery()})`, 'subquery')
       .setParameter('path', itemPath)
       .setParameter('view', view)
-      .setParameter('types', types);
+      .setParameter('types', types)
+      .setParameter('startDate', filters?.startDate)
+      .setParameter('endDate', filters?.endDate);
```

Comment:
I am not sure you need to set these parameters ? Looks like you pass them already in the `andWhere`.

Reply:
It didn't work unless I passed them